### PR TITLE
Cache `class_is_abstract()`

### PR DESCRIPTION
**Pull request by jacobtylerwalls**
## Type of Changes

|     | Type                   |
| --- | ---------------------- |
| ✓   | :hammer: Refactoring   |

## Description
For one project in the pylint primer (music21), provides 15% speedup _linting the entire project_. (171.6s -> 145.6s on an M1 mac running python3.11.)

I think we should batch the performance improvements in 3.0 and describe them all with a set of benchmarks at the end, so adding a skip news label.

### CODE REVIEW

**Review by Pierre-Sassoulas — COMMENTED** [review]
How about we create the entry right now and add an entry each time we do a performance fix ? This way we're not going to have to do it all at once and each MR can verify that a new detail is added.

File: towncrier.toml [hunk 1]
```diff
@@ -65,3 +65,8 @@ showcontent = true
 directory = "internal"
 name = "Internal Changes"
 showcontent = true
+
+[[tool.towncrier.type]]
+directory = "performance"
+name = "Performance Improvements"
+showcontent = true
```
Comment (Pierre-Sassoulas) [inline towncrier.toml:72]: 👍 but the doc and issue template needs to be updated for this.

Comment (jacobtylerwalls) [reply towncrier.toml:72]: No problem, but not sure where to look. Were you thinking of the PR template?

Comment (Pierre-Sassoulas) [reply towncrier.toml:72]: Yes but nvm we changed it to this recently: https://github.com/pylint-dev/pylint/blob/main/.github/PULL_REQUEST_TEMPLATE.md?plain=1#L9

**Review by Pierre-Sassoulas — APPROVED** [review]

**Pull request merged**